File: articles/linear-algebra-for-systems-modeling/articles/interpretation-approximation-and-responsible-mathematical-modeling/calculators/python/responsible_modeling_calculator.py
from __future__ import annotations

import csv
import json
from pathlib import Path


def main() -> None:
    output_dir = Path("outputs")
    output_dir.mkdir(parents=True, exist_ok=True)

    result = {
        "calculator": "interpretation_approximation_and_responsible_mathematical_modeling_calculator",
        "workflow_name": "responsible_modeling_audit",
        "model_purpose": "interpret_linear_algebra_output_for_systems_modeling",
        "claim_type": "exploratory_decision_support_not_causal_proof",
        "approximation_form": "linear_or_low_rank_approximation_with_explicit_assumptions",
        "validation_status": "validated_only_for_stated_data_range_operating_context_and_model_purpose",
        "interpretation_boundary": "Outputs support structured interpretation within the stated assumptions, not universal claims, causal proof, or unreviewed decision authority.",
        "warning": "Model use requires documented assumptions, validation evidence, review status, uncertainty communication, and stop-use conditions."
    }

    with (output_dir / "interpretation_approximation_and_responsible_mathematical_modeling_calculator.json").open("w", encoding="utf-8") as handle:
        json.dump(result, handle, indent=2, sort_keys=True)

    with (output_dir / "interpretation_approximation_and_responsible_mathematical_modeling_calculator.csv").open("w", newline="", encoding="utf-8") as handle:
        writer = csv.DictWriter(handle, fieldnames=list(result.keys()))
        writer.writeheader()
        writer.writerow(result)

    print(json.dumps(result, indent=2, sort_keys=True))


if __name__ == "__main__":
    main()

File: articles/linear-algebra-for-systems-modeling/articles/interpretation-approximation-and-responsible-mathematical-modeling/calculators/r/responsible_modeling_calculator.R
result <- data.frame(
  calculator = "interpretation_approximation_and_responsible_mathematical_modeling_calculator",
  workflow_name = "responsible_modeling_audit",
  model_purpose = "interpret_linear_algebra_output_for_systems_modeling",
  claim_type = "exploratory_decision_support_not_causal_proof",
  approximation_form = "linear_or_low_rank_approximation_with_explicit_assumptions",
  validation_status = "validated_only_for_stated_data_range_operating_context_and_model_purpose",
  interpretation_boundary = "Outputs support structured interpretation within the stated assumptions, not universal claims, causal proof, or unreviewed decision authority.",
  warning = "Model use requires documented assumptions, validation evidence, review status, uncertainty communication, and stop-use conditions."
)

dir.create("outputs", recursive = TRUE, showWarnings = FALSE)
write.csv(result, "outputs/r_interpretation_approximation_and_responsible_mathematical_modeling_calculator.csv", row.names = FALSE)
print(result)
